pandas/tests/indexes/datetimes/test_partial_slicing.py

```diff
@@ -330,3 +332,23 @@ def test_loc_datetime_length_one(self):
 
         result = df.loc['2016-10-01T00:00:00':]
         tm.assert_frame_equal(result, df)
+
+    comparisonTS = Timestamp('20130101')
+
+    @pytest.mark.parametrize('datetimelike,opExp', product(
+        [comparisonTS, comparisonTS.to_pydatetime(),
+         comparisonTS.to_pydatetime().date(), comparisonTS.to_datetime64()],
+        [(op.lt, [True, False, False, False]),
```

Comment:
don't pass ``opExp``, rather pass ``op`` (e.g. the op.lt) and the ``expected`` (the boolean result)

Reply:
Can do, but i am not sure something like:
```
[(x, y[0], y[1]) for x, y in product( 
    [pd.Timestamp('20130101'), datetime.datetime(2013, 1, 1),
     datetime.date(2013, 1, 1), np.datetime64("2013-01-01", 'ns')],
    [(op.lt, [True, False, False, False])])]
```
is more readable.

Reply:
sorry didn't see the product. instead use multiple paramatrize to achieve product.

eg..
```
@pytest.mark.parametrize(.....)
@pytest.mark.parametrize('datetimelike'
```